plugins/org.jkiss.dbeaver.ext.informix/src/org/jkiss/dbeaver/ext/informix/model/InformixTrigger.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2021 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.informix.model;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableBase;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableTrigger;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.utils.CommonUtils;
+
+public class InformixTrigger extends GenericTableTrigger {
+
+    public enum TriggerEventType {
+        D("DELETE"),
+        I("INSERT"),
+        U("UPDATE"),
+        S("SELECT"),
+        d("INSTEAD OF Delete"),
+        i("INSTEAD OF Insert"),
+        u("INSTEAD OF Update");
+
+        private final String eventType;
+
+        TriggerEventType(String eventType) {
+            this.eventType = eventType;
+        }
+
+        public String getEventType() {
+            return eventType;
+        }
+    }
+
+    private TriggerEventType eventType;
+    private final String beforeValue;
+    private final String afterValue;
+    private final String collation;
+
+    public InformixTrigger(@NotNull GenericTableBase container, String name, @NotNull JDBCResultSet resultSet) {
+        super(container, name, null);
+        String eventTypeLetter = JDBCUtils.safeGetString(resultSet, "event");
+        if (CommonUtils.isNotEmpty(eventTypeLetter)) {
+            this.eventType = TriggerEventType.valueOf(eventTypeLetter);
+        }
```

Comment:
This may lead to unchecked exceptions appearing all over the place. Use `CommonUtils.valueOf` instead with supplied default value.

Reply:
Not sure about the default value for Event. Null is the best default value for this situation :)

Reply:
You can specify `null` as default value :^)